unblob/handlers/compression/lzo.py

```diff
@@ -0,0 +1,98 @@
+import io
+from typing import List, Optional
+
+from structlog import get_logger
+
+from ...models import StructHandler, ValidChunk
+
+logger = get_logger()
+
+
+F_H_FILTER = 0x00000800
+
+
+class LZOHandler(StructHandler):
+    NAME = "lzo"
+
+    YARA_RULE = r"""
+        strings:
+            $lzo_magic = { 89 4C 5A 4F 00 0D 0A 1A 0A }
+        condition:
+            $lzo_magic
+    """
+
+    C_DEFINITIONS = r"""
+        struct lzo_header_no_filter
+        {
+            char magic[9];
+            uint16 version;
+            uint16 libversion;
+            uint16 reqversion;
+            uint8 method;
+            uint8 level;
+            uint32 flags;
+            //uint32 filter;              // only if flags & F_H_FILTER
+            uint32 mode;
+            uint32 mtime;
+            uint32 gmtdiff;
+            uint8 filename_len;
+            //char filename[];
```

Comment:
I kept it here for documentation but it would be better to have a single line comment explaining why. The filename length is variable and set by `filename_len`, so we don't know what's the exact filename char array length at parsing time. That's why it is handled dynamically at https://github.com/IoT-Inspector/unblob/pull/89/files#diff-9c572eb32097246fa6c46bd103c604618f9c899900cc72215f40d1e486511aeeR79

Reply:
Switched to clear inline comment.